temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowRunTaskHandler.java

```diff
@@ -255,10 +255,14 @@ private void applyServerHistory(long lastEventId, WorkflowHistoryIterator histor
               implementationOptions.getFailWorkflowExceptionTypes();
           for (Class<? extends Throwable> failType : failTypes) {
             if (failType.isAssignableFrom(e.getClass())) {
+              metricsScope.counter(MetricsType.WORKFLOW_FAILED_COUNTER).inc(1);
               throw new WorkflowExecutionException(
                   workflow.getWorkflowContext().mapWorkflowExceptionToFailure(e));
             }
           }
+          if (e instanceof WorkflowExecutionException) {
```

Comment:
Curious, what is the situation where `e` is a `WorkflowExecutionException` here?

Reply:
I don't think it can happen today, but if `workflowStateMachines.handleEvent(event, hasNext);` did throw a `WorkflowExecutionException` it would lead to a workflow failure. So it is more future proofing, it also may be possible with some user interceptor could throw a `WorkflowExecutionException`.

Reply:
Works for me.